src/custom-ui/CustomWidgets.js

```diff
@@ -66,6 +66,15 @@ const CustomTextWidget = (props) => {
     readonly={true}
     value={schema.const}
   />
+  } else if (schema.title === 'decimal') {
+    const onChange = (val) => {
+      if (val === undefined || val.match(/^-?\d{0,}(\.\d{0,})?$/)) {
```

Comment:
This regex does not match the one used in schemaHandler.js
```suggestion
      if (val === undefined || val.match(/^-?\d{1,}(\.\d{0,})?$/)) {
```

Reply:
Yes it needs to not match in order top allow the user to type in a value. So this regex allows a user to type just '-' as a start to a number but the validator will complain if you try and submit just '-'. Same with the decimal place. User needs to be able to type '2.'  as a start but '2.' will not be valid.

Reply:
I see, that makes sense! Then the ".7" test will need to be updated to allow ".7". It is failing now.